src/lib/custom-deploy/custom-deploy.ts

```diff
@@ -13,47 +13,74 @@ import { validateDeployFileExt, validateFilename } from '../../lib/validations/c
 const DEPLOY_MAX_FILE_SIZE = 4 * GB_IN_BYTES;
 const WPVIP_DEPLOY_TOKEN = process.env.WPVIP_DEPLOY_TOKEN;
 
+type CustomDeployInfo = {
+	success: boolean;
+	appId: number;
+	envId: number;
+	envType: string;
+	primaryDomainName: string;
+	launched: boolean;
+	customDeploysEnabled: boolean;
+};
+
+type ValidateMutationPayload = {
+	data?: {
+		validateCustomDeployAccess: CustomDeployInfo;
+	} | null;
+};
+
 export function isSupportedApp( app: App ): boolean {
 	return WORDPRESS_SITE_TYPE_IDS.includes( app.typeId as number );
 }
 
-export async function validateCustomDeployKey( envId: number ): Promise< void > {
+export async function validateCustomDeployKey(
+	app: string | number,
+	env: string | number
+): Promise< CustomDeployInfo > {
 	if ( ! WPVIP_DEPLOY_TOKEN ) {
 		exit.withError( 'Valid custom deploy key is required.' );
 	}
 
 	const VALIDATE_CUSTOM_DEPLOY_ACCESS_MUTATION = gql`
 	mutation ValidateCustomDeployAccess {
-		validateCustomDeployAccess( input: { environmentIds: ${ envId } } ) {
-			success
+		validateCustomDeployAccess( input: { app: "${ String( app ) }", env: "${ String( env ) }" } ) {
+			success,
+			appId,
+			envId,
+			envType,
+			primaryDomainName,
+			launched,
+			customDeploysEnabled
 		}
 	}
 `;
 
 	const api = API();
 	try {
-		await api.mutate( {
+		const result: ValidateMutationPayload = await api.mutate( {
 			mutation: VALIDATE_CUSTOM_DEPLOY_ACCESS_MUTATION,
 			context: {
 				headers: {
 					Authorization: `Bearer ${ WPVIP_DEPLOY_TOKEN }`,
 				},
 			},
 		} );
+
+		if ( ! result.data?.validateCustomDeployAccess ) {
+			throw new Error( 'Not found' );
```

Comment:
Need a better error descriptor

Reply:
This one is just to trigger the catch block, so the message here never gets passed to the user.